ui/src/shared/components/FunctionSelector.js

```diff
@@ -48,47 +54,65 @@ class FunctionSelector extends Component {
 
   render() {
     const {localSelectedItems} = this.state
+    const {singleSelect} = this.props
 
     return (
       <div className="function-selector">
-        <div className="function-selector--header">
-          <span>
-            {localSelectedItems.length > 0
-              ? `${localSelectedItems.length} Selected`
-              : 'Select functions below'}
-          </span>
-          <div
-            className="btn btn-xs btn-success"
-            onClick={this.handleApplyFunctions}
-          >
-            Apply
-          </div>
-        </div>
-        <div className="function-selector--grid">
-          {INFLUXQL_FUNCTIONS.map((f, i) => {
-            return (
+        {singleSelect
+          ? null
+          : <div className="function-selector--header">
+              <span>
+                {localSelectedItems.length > 0
+                  ? `${localSelectedItems.length} Selected`
+                  : 'Select functions below'}
+              </span>
               <div
-                key={i}
-                className={classnames('function-selector--item', {
-                  active: this.isSelected(f),
-                })}
-                onClick={_.wrap(f, this.onSelect)}
+                className="btn btn-xs btn-success"
+                onClick={this.handleApplyFunctions}
               >
-                {f}
+                Apply
               </div>
-            )
-          })}
+            </div>}
+        <div className="function-selector--grid">
+          {singleSelect
+            ? INFLUXQL_FUNCTIONS.map((f, i) => {
+                return (
+                  <div
+                    key={i}
+                    className={classnames('function-selector--item', {
+                      active: this.isSelected(f),
+                    })}
+                    onClick={_.wrap(f, this.onSingleSelect)}
+                  >
+                    {f}
+                  </div>
+                )
+              })
+            : INFLUXQL_FUNCTIONS.map((f, i) => {
+                return (
+                  <div
+                    key={i}
+                    className={classnames('function-selector--item', {
+                      active: this.isSelected(f),
+                    })}
+                    onClick={_.wrap(f, this.onSelect)}
+                  >
+                    {f}
+                  </div>
+                )
+              })}
```

Comment:
It looks like these two maps do the same thing, but all they need are different functions passed to the onClick event handler.